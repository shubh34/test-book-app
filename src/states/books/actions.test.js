import {
  fetchBooks,
  fetchBookRequest,
  fetchBookSuccess,
  fetchBookFailure,
  FETCH_BOOKS_REQUEST,
  FETCH_BOOKS_SUCCESS,
  FETCH_BOOKS_FAILURE
} from './actions';

describe('Books actions', () => {
  describe('fetchBooks', () => {
    it('should create action for fetchBooks with default params', () => {
      expect(fetchBooks()).toMatchSnapshot();
    });
    it('should create action for fetchBooks with passed params', () => {
      expect(
        fetchBooks({
          page: 2,
          itemsPerPage: 60,
          filters: [{ type: 'all', values: ['YOUR_SEARCH_FIELD_CONTENTS'] }]
        })
      ).toMatchSnapshot();
    });
  });
  describe('fetchBookRequest', () => {
    it('should create action for fetch book request', () => {
      expect(fetchBookRequest(1)).toEqual({
        type: FETCH_BOOKS_REQUEST,
        meta: { page: 1 }
      });
    });
  });
  describe('fetchBookSuccess', () => {
    it('should create action for fetchBooks success', () => {
      expect(fetchBookSuccess(1)).toEqual({
        type: FETCH_BOOKS_SUCCESS,
        meta: { page: 1 }
      });
    });
  });
  describe('fetchBookFailure', () => {
    it('should create action for fetchBooks failure', () => {
      expect(fetchBookFailure(1)).toEqual({
        type: FETCH_BOOKS_FAILURE,
        meta: { page: 1 }
      });
    });
  });
});
