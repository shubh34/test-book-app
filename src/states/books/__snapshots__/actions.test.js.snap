// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Books actions should create action for fetchBooks with default params 1`] = `
Object {
  "@@redux-api-middleware/RSAA": Object {
    "body": Object {
      "filters": Array [],
      "itemsPerPage": 20,
      "page": 1,
    },
    "endpoint": "/api/books",
    "method": "POST",
    "types": Array [
      "GET_BOOKS_REQUEST",
      "GET_BOOKS_SUCCESS",
      "GET_BOOKS_FAILURE",
    ],
  },
}
`;

exports[`Books actions should create action for fetchBooks with passed params 1`] = `
Object {
  "@@redux-api-middleware/RSAA": Object {
    "body": Object {
      "filters": Array [
        Object {
          "type": "all",
          "values": Array [
            "YOUR_SEARCH_FIELD_CONTENTS",
          ],
        },
      ],
      "itemsPerPage": 60,
      "page": 2,
    },
    "endpoint": "/api/books",
    "method": "POST",
    "types": Array [
      "GET_BOOKS_REQUEST",
      "GET_BOOKS_SUCCESS",
      "GET_BOOKS_FAILURE",
    ],
  },
}
`;
